source/hwIo.py

```diff
@@ -312,6 +312,16 @@ def setFeature(self, report):
 				log.debug("Set feature failed: %s" % ctypes.WinError())
 			raise ctypes.WinError()
 
+	def setOutputReport(self,data):
+		"""
+		Write the given data to the device using HidD_SetOutputReport.
+		This is instead of using the standard WriteFile which may freeze with some USB HID implementations.
+		@ param data: the report string (starting with the report ID).
+		"""
+		buf=ctypes.create_string_buffer(self._writeSize)

```

Comment:
In setFeature, the buffer is as long as the report. I guess using writeSize just works, but I think using the length of the data instead deserves a try, since in many cases, especially with the BI 14, it involves less bytes to send.